cfme/tests/infrastructure/test_host.py

```diff
@@ -469,6 +470,44 @@ def test_compare_hosts_from_provider_allhosts(appliance, setup_provider_min_host
     assert compare_hosts_view.is_displayed
 
 
+@test_requirements.infra_hosts
+@pytest.mark.meta(blockers=[BZ(1747545, forced_streams=["5.10"])], automates=[1747545])
+@pytest.mark.parametrize("num_hosts", [2, 4])
+@pytest.mark.parametrize("hosts_collection", ["provider", "appliance"])
+@pytest.mark.parametrize(
+    "report_format", ["Download as Text", "Download as CSV", "Print or export as PDF"],
```

Comment:
I wouldn't define these strings here, as it will become troublesome if we have to version pick the available options, or the strings for a given version.

It is a common list through MIQ pages that include download, it would be nice to define in a way that's easy to utilize outside of just infra hosts view classes.

I'm thinking, create a NamedConstant class within IPAppliance class?

```
class IPAppliance():
    class DownloadOptions(aenum.NamedConstant):
        TEXT = 'Download as Text'

## In use
view.toolbar.download.item_select(str(appliance.DownloadOptions.TEXT))
```

Reply:
I see this same parametrization is used above, for the same testcase but not from the compare view.

There is a lot of test function and parametrization duplication in this test module, in order to test just slightly different behaviors.  

Would be nice to just combine this with the existing test functions above, as a parameter of the test function, or just in addition to the basic compare test.

Reply:
I am working on adding and testing the NamedConstant class. I do agree that we should combine the hosts test functionality into fewer test functions. I'd prefer to get this test merged and have the coverage, and I can work on combining tests afterward. I'll be able analyze our coverage, put together some plans for combining the tests, and get team input before I finialize the coding.

Reply:
I created DownloadOptions, but still have some questions. Should I use "PRINT" or "PDF" for the 'Print or export to PDF' option? Will my implementation of report_format parameterization work? Is there an appropriate way to parameterize with objects instead of text (using another fixture perhaps)?